consensus/src/creation/mod.rs

```diff
@@ -51,20 +51,28 @@ pub struct IO<H: Hasher> {
     pub(crate) outgoing_units: Sender<NotificationOut<H>>,
 }
 
+fn very_long_delay() -> Delay {
+    Delay::new(Duration::from_secs(30 * 60))
+}
+
 async fn create_unit<H: Hasher>(
     round: Round,
     creator: &mut Creator<H>,
-    create_lag: &DelaySchedule,
-    mut can_create: bool,
+    delay: Option<Delay>,
     incoming_parents: &mut Receiver<Unit<H>>,
     mut exit: &mut oneshot::Receiver<()>,
 ) -> Result<(PreUnit<H>, Vec<H::Hash>), ()> {
-    let mut delay = Delay::new(create_lag(round.into())).fuse();
+    let (initial_delay, mut delay_passed) = match delay {
+        None => (very_long_delay(), true),
```

Comment:
I don't like it at all now, it's longer and more confusing than before.
Two bools and computing the delay inside this function is better.
The best is imho to stop pretending that avoiding code duplication with this artificial delay is cool. Just make two similar functions, one for the case when we wait for the delay, or consider two cases with very similar `select`s.

Reply:
See the comment above.